<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>595</width>
    <height>619</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Neural network UI</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>577</width>
      <height>491</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="trainAETab">
     <attribute name="title">
      <string>Entraîner un auto-encodeur</string>
     </attribute>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0" colspan="3">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Emplacement du dataset (liste des fichiers)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="datasetPathLineEdit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="datasetPathButton">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0" colspan="3">
       <widget class="QTabWidget" name="tabWidget_2">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Options d'entraînement</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="3" column="1">
           <widget class="QComboBox" name="optimizerCombobox">
            <property name="currentText">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="trainingEpochCount">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Optimiseur :</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Taille du batch :</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="trainingBatchSize">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>64</number>
            </property>
            <property name="value">
             <number>4</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Nombre d'époques :</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="3">
           <widget class="QTabWidget" name="optimizerSelectTab">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="AdamTab">
             <attribute name="title">
              <string>Adam</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="0" column="0">
               <widget class="QScrollArea" name="scrollArea">
                <property name="widgetResizable">
                 <bool>true</bool>
                </property>
                <widget class="QWidget" name="scrollAreaWidgetContents">
                 <property name="geometry">
                  <rect>
                   <x>0</x>
                   <y>0</y>
                   <width>503</width>
                   <height>188</height>
                  </rect>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_5">
                  <item row="0" column="1">
                   <widget class="QDoubleSpinBox" name="AdamLearningRate">
                    <property name="decimals">
                     <number>9</number>
                    </property>
                    <property name="maximum">
                     <double>1.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.000100000000000</double>
                    </property>
                    <property name="stepType">
                     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                    </property>
                    <property name="value">
                     <double>0.000100000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>Learning rate :</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="2">
                   <spacer name="horizontalSpacer_3">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>40</width>
                      <height>20</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                  <item row="1" column="0" colspan="3">
                   <spacer name="verticalSpacer">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>40</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </widget>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="ASGDTab">
             <attribute name="title">
              <string>ASGD</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_7">
              <item row="0" column="0">
               <widget class="QScrollArea" name="scrollArea_2">
                <property name="widgetResizable">
                 <bool>true</bool>
                </property>
                <widget class="QWidget" name="scrollAreaWidgetContents_2">
                 <property name="geometry">
                  <rect>
                   <x>0</x>
                   <y>0</y>
                   <width>503</width>
                   <height>162</height>
                  </rect>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_6">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_7">
                    <property name="text">
                     <string>Learning rate :</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_8">
                    <property name="text">
                     <string>Decay :</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QDoubleSpinBox" name="ASGDLearningRate">
                    <property name="decimals">
                     <number>9</number>
                    </property>
                    <property name="maximum">
                     <double>1.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.000100000000000</double>
                    </property>
                    <property name="stepType">
                     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                    </property>
                    <property name="value">
                     <double>0.000100000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="2">
                   <spacer name="horizontalSpacer_4">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>40</width>
                      <height>20</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                  <item row="1" column="1">
                   <widget class="QDoubleSpinBox" name="ASGDlambd">
                    <property name="decimals">
                     <number>9</number>
                    </property>
                    <property name="maximum">
                     <double>1.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.000100000000000</double>
                    </property>
                    <property name="stepType">
                     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                    </property>
                    <property name="value">
                     <double>0.000100000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="0" colspan="3">
                   <spacer name="verticalSpacer_2">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>40</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </widget>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="RMSPropTab">
             <attribute name="title">
              <string>RMSProp</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_9" rowstretch="0">
              <item row="0" column="0">
               <widget class="QScrollArea" name="scrollArea_3">
                <property name="widgetResizable">
                 <bool>true</bool>
                </property>
                <widget class="QWidget" name="scrollAreaWidgetContents_3">
                 <property name="geometry">
                  <rect>
                   <x>0</x>
                   <y>0</y>
                   <width>503</width>
                   <height>162</height>
                  </rect>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_8">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_9">
                    <property name="text">
                     <string>Learning rate :</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1" colspan="2">
                   <widget class="QDoubleSpinBox" name="RMSPropLearningRate">
                    <property name="decimals">
                     <number>9</number>
                    </property>
                    <property name="maximum">
                     <double>1.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.000100000000000</double>
                    </property>
                    <property name="stepType">
                     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                    </property>
                    <property name="value">
                     <double>0.000100000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_10">
                    <property name="text">
                     <string>Momentum :</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1" colspan="2">
                   <widget class="QDoubleSpinBox" name="RMSPropMomentum">
                    <property name="decimals">
                     <number>9</number>
                    </property>
                    <property name="maximum">
                     <double>1.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.000100000000000</double>
                    </property>
                    <property name="stepType">
                     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                    </property>
                    <property name="value">
                     <double>0.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="0" colspan="2">
                   <widget class="QLabel" name="label_11">
                    <property name="text">
                     <string>Smoothing constant :</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="2">
                   <widget class="QDoubleSpinBox" name="RMSPropAlpha">
                    <property name="decimals">
                     <number>9</number>
                    </property>
                    <property name="maximum">
                     <double>1.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.000100000000000</double>
                    </property>
                    <property name="stepType">
                     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                    </property>
                    <property name="value">
                     <double>0.990000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="0">
                   <widget class="QCheckBox" name="RMSPropCentered">
                    <property name="text">
                     <string>Centered</string>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="3">
                   <spacer name="horizontalSpacer_5">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>40</width>
                      <height>20</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                  <item row="4" column="1" colspan="2">
                   <spacer name="verticalSpacer_3">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>37</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </widget>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item row="2" column="2">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Options de dataset</string>
         </attribute>
        </widget>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Nom du fichier de destination</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="newAutoEncoderNameLineEdit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="newAutoEncoderSaveButton">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="3" column="1" colspan="2">
       <widget class="QPushButton" name="trainAutoEncoderButton">
        <property name="text">
         <string>Entraîner l'autoencodeur</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>595</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
